### PR TITLE
TDI-38353:NPE exception happen when 'Azure Shared Access Signature'

### PR DESCRIPTION
**Pull request by pyzhou-talend**
input a invalid value.
https://jira.talendforge.org/browse/TDI-38353
**Please check if the PR fulfills these requirements**

- [x] The commit message follows Talend standard
- [ ] Tests for the changes have been added (for bug fixes / features)

**What is the current behavior?** (You can also link to an open issue here)

**What is the new behavior?**

**Does this PR introduce a breaking change?**

- [ ] Yes
- [x] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

File: components/components-azurestorage/src/main/java/org/talend/components/azurestorage/utils/SharedAccessSignatureUtils.java [hunk 1]
```diff
@@ -38,12 +39,12 @@ public SharedAccessSignatureUtils(String sas, String protocol, String account, S
         this.token = sap;
     }
 
-    public static SharedAccessSignatureUtils getSharedAccessSignatureUtils(String sas) {
+    public static SharedAccessSignatureUtils getSharedAccessSignatureUtils(String sas) throws InvalidKeyException {
```
Comment (pteyssier) [inline components/components-azurestorage/src/main/java/org/talend/components/azurestorage/utils/SharedAccessSignatureUtils.java]: I prefer you use IllegalParameterException here

Comment (pteyssier) [inline components/components-azurestorage/src/main/java/org/talend/components/azurestorage/utils/SharedAccessSignatureUtils.java]: My message should not be on that line but below

**Comment by build-talend-common** [comment]
Pull request test result: [failure](https://ci-common.datapwn.com/job/components_pull-request_master/208/) [:x:](https://ci-common.datapwn.com/job/components_pull-request_master/208/)

**Comment by build-talend-common** [comment]
Pull request test result: [failure](https://ci-common.datapwn.com/job/components_pull-request_master/236/) [:x:](https://ci-common.datapwn.com/job/components_pull-request_master/236/)

**Comment by build-talend-common** [comment]
Pull request test result: [failure](https://ci-common.datapwn.com/job/components_pull-request_master/237/) [:x:](https://ci-common.datapwn.com/job/components_pull-request_master/237/)

**Comment by build-talend-common** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/components_pull-request_master/241/) [:star:](https://ci-common.datapwn.com/job/components_pull-request_master/241/)

File: components/components-azurestorage/src/main/java/org/talend/components/azurestorage/utils/SharedAccessSignatureUtils.java [hunk 2]
```diff
@@ -38,12 +38,12 @@ public SharedAccessSignatureUtils(String sas, String protocol, String account, S
...
     }
 
-    public static SharedAccessSignatureUtils getSharedAccessSignatureUtils(String sas) {
+    public static SharedAccessSignatureUtils getSharedAccessSignatureUtils(String sas){
         Matcher m = Pattern.compile(SharedAccessSignatureUtils.SAS_PATTERN).matcher(sas);
         if (m.matches()) {
             return new SharedAccessSignatureUtils(sas, m.group(1), m.group(2), m.group(3), m.group(4));
         }
-        return null;
+        throw new IllegalArgumentException("Shared Acces Signature is invalid.");
```
Comment (gonchar-ivan) [inline components/components-azurestorage/src/main/java/org/talend/components/azurestorage/utils/SharedAccessSignatureUtils.java]: I consider 2 types of exceptional situations in our components:
1. Caused by wrong usage of method and the problem was caused by component developer. It this case we may produce exception with English only messages
2. Caused by user, when he provides wrong properties values. In this case message should be i18n.

Comment (pyzhou-talend) [inline components/components-azurestorage/src/main/java/org/talend/components/azurestorage/utils/SharedAccessSignatureUtils.java]: Yes. you are right. However I have another issue for i18n. https://jira.talendforge.org/browse/TDI-38333  . I think I will make i18n in that issue.

Comment (gonchar-ivan) [inline components/components-azurestorage/src/main/java/org/talend/components/azurestorage/utils/SharedAccessSignatureUtils.java]: ok for me to fix it in separate issue. Already approved PR.
I think you also need approval from @pteyssier

**Pull request merged**